ultet/src/main/scala/za/co/absa/ultet/dbitems/DBSchema.scala

```diff
@@ -16,17 +16,38 @@
 
 package za.co.absa.ultet.dbitems
 
+import com.typesafe.scalalogging.Logger
+import za.co.absa.ultet.dbitems.DBSchema.{DO_NOT_CHOWN, DO_NOT_TOUCH, logger}
 import za.co.absa.ultet.model.{SQLEntry, SchemaName, UserName}
 import za.co.absa.ultet.model.schema.{SchemaCreate, SchemaGrant, SchemaOwner}
 
 case class DBSchema(name: SchemaName,
                     ownerName: UserName,
                     users: Seq[UserName]) extends DBItem {
   override def sqlEntries: Seq[SQLEntry] = {
-    Seq(
-      SchemaCreate(name),
-      SchemaOwner(name, ownerName),
-      SchemaGrant(name, users)
-    )
+    if (DO_NOT_TOUCH.contains(name.value)) {
+      throw new Exception(s"Schema ${name.value} is not allow to be referenced")
+    }
+
+    if (DO_NOT_CHOWN.contains(name.value)){
+      logger.warn(s"Schema ${name.value} is not allowed to change owner")
+      Seq(
+        SchemaCreate(name),
+        SchemaGrant(name, users)
+      )
+    } else {
+      Seq(
+        SchemaCreate(name),
+        SchemaOwner(name, ownerName),
+        SchemaGrant(name, users)
+      )
+    }
   }
 }
+
+object DBSchema{
+  private val logger = Logger(getClass.getName)
+
+  val DO_NOT_TOUCH: Seq[String] = Seq("pg_toast", "pg_catalog", "information_schema")
+  val DO_NOT_CHOWN: Seq[String] = Seq("public")
```

Comment:
I don't think that these vals conform to Scala naming conventions. Perhaps `DoNotTouch` etc?